src/scales/scale.linearbase.js

```diff
@@ -44,6 +44,13 @@ function generateTicks(generationOptions, dataRange) {
 	niceMin = Math.floor(dataRange.min / spacing) * spacing;
 	niceMax = Math.ceil(dataRange.max / spacing) * spacing;
 
+	if (niceMin === niceMax) {
+		// This happens due to float inccuracy when min and max are really close to each other
+		// One case: min = 1.8548483304974972 and max = 1.8548483304974974
+		ticks.push(niceMin - helpers.EPSILON, niceMax, niceMax + helpers.EPSILON);
```

Comment:
Can we instead do:

    ticks.push(dataRange.min, dataRange.max)

`niceMin - helpers.EPSILON` and `niceMax + helpers.EPSILON` can appear outside the range of any of the data points for no reason, which can be a bit strange. I think it would be nicer to just use the actual data instead of rounding and adding an epsilon

I'm also wondering what would happen if we instead changed this to:
```
if (niceMin === niceMax) {
  niceMin = dataRange.min;
  niceMax = dataRange.max;
}
```

If we push only 2 or 3 ticks that might not be enough in all cases. The fiddle doesn't demonstrate it since it has only 2 data points, but what if you have a lot of points in a very small floating point range? It might be a better solution to just skip the rounding logic in the case that we have small floating point numbers. I think we may still want to generate multiple ticks, just without so much rounding

Reply:
> Can we instead do:
> 
> ```
> ticks.push(dataRange.min, dataRange.max)
> ```

Yes we can, good catch!
![image](https://user-images.githubusercontent.com/27971921/50655838-054fb200-0f9a-11e9-829a-2ba037ec8b0b.png)


> `niceMin - helpers.EPSILON` and `niceMax + helpers.EPSILON` can appear outside the range of any of the data points for no reason, which can be a bit strange. I think it would be nicer to just use the actual data instead of rounding and adding an epsilon
> 
> I'm also wondering what would happen if we instead changed this to:
> 
> ```
> if (niceMin === niceMax) {
>   niceMin = dataRange.min;
>   niceMax = dataRange.max;
> }
> ```

This one ends up with strange values at labels
![image](https://user-images.githubusercontent.com/27971921/50655566-44c9ce80-0f99-11e9-9a28-ac5c275b4ca4.png)

> If we push only 2 or 3 ticks that might not be enough in all cases. The fiddle doesn't demonstrate it since it has only 2 data points, but what if you have a lot of points in a very small floating point range? It might be a better solution to just skip the rounding logic in the case that we have small floating point numbers. I think we may still want to generate multiple ticks, just without so much rounding

Those points must be in such a small range (within EPSILON from each other), so there is no reason to draw more than 2 ticks.

For example with this dataset: 
```
                  {x: 1, y: 1.8548483304974972},
                  {x: 2, y: 1.8548483304974974},
                  {x: 3, y: 1.8548483304974975},
                  {x: 4, y: 1.8548483304974973}
```
![image](https://user-images.githubusercontent.com/27971921/50656273-47c5be80-0f9b-11e9-8c7c-d6397678ad44.png)

```
var x=1.8548483304974973
x
1.8548483304974972
```

Reply:
Is there some kind of rounding in the plotting that's making the first and fourth points show up at the same value and the second and third points show up at the same value?

Reply:
> Is there some kind of rounding in the plotting that's making the first and fourth points show up at the same value and the second and third points show up at the same value?

`float` happens there, those ending at 3 or 5 can not be accurately presented with floating points.